### PR TITLE
Add tests for user interruptions

### LINKED ISSUE #5233: Test the user interruptions manager
Test the user interruptions manager, including its default and customized behavior.

### LINKED ISSUE #5235: Test the `AsyncMutex`
Test the AsyncMutex under `./v-next/core`. The mutex is used by the user interruptions manager.

> [!NOTE]  
> time box this, it may be too challenging or even impossible.

### PR DESCRIPTION
**Pull request by zoeyTM**
resolves #5233 
resolves #5235 

added tests for `UserInterruptionsManagerImplementation` and `AsyncMutex`

### CODE REVIEW

File: v-next/core/test/internal/user-interruptions/async-mutex.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      running--;
+    }
+
+    await Promise.all([
+      mutex.excluiveRun(run),
+      mutex.excluiveRun(run),
+      mutex.excluiveRun(run),
+    ]);
+
+    assert.strictEqual(maxRunning, 1);
```
Comment (kanej) [inline v-next/core/test/internal/user-interruptions/async-mutex.ts:26]: Lets assert running is 0 as well.

File: v-next/core/test/internal/user-interruptions/async-mutex.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { AsyncMutex } from "../../../src/internal/async-mutex.js";
+
+describe("AsyncMutex", () => {
+  it("should run a function exclusively", async () => {
+    const mutex = new AsyncMutex();
+
+    let running = 0;
+    let maxRunning = 0;
+
+    async function run() {
```
Comment (schaable) [inline v-next/core/test/internal/user-interruptions/async-mutex.ts:13]: nice!

File: v-next/core/test/internal/user-interruptions/user-interruptions-manager.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      };
+
+      hookManager.registerHandlers("userInterruptions", handlers);
+
+      await userInterruptionManager.displayMessage(
+        "test-interruptor",
+        "test-message",
+      );
+
+      assert(called);
```
Comment (schaable) [inline v-next/core/test/internal/user-interruptions/user-interruptions-manager.ts:36]: It's better if we add a `message` to `assert`, as it's an alias of `assert.ok` and there's a bug that makes `assert.ok` really slow unless you provide a `message`. See: https://github.com/nodejs/node/issues/52962

**Review by schaable — APPROVED** [review]
Just a minor comment, but LGTM

File: v-next/core/test/internal/user-interruptions/async-mutex.ts (hunk 2 above)
Comment (kanej) [reply v-next/core/test/internal/user-interruptions/async-mutex.ts:13]: Agreed.

File: v-next/core/package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "eslint": "eslint \"src/**/*.ts\" \"test/**/*.ts\"",
     "prettier": "prettier \"**/*.{ts,js,md,json}\"",
     "test": "glob --cmd=\"node --import tsx/esm --test\" \"test/**/*.ts\"",
+    "test:only": "glob --cmd=\"node --import tsx/esm --test --test-only\" \"test/**/*.ts\"",
```
Comment (kanej) [inline v-next/core/package.json:36]: I would drop this line and let the later PR add it.

**Review by kanej — APPROVED** [review]
Looks good. Well done on the mutex test.

There are two points I am interested in:

- is there value in testing around `uninterrupted`?
- what do we think about tests of the default implementations?

I think we merge this PR and depending on the discussion, follow up on these two points.

**Comment by zoeyTM** [comment]
@kanej 

- I don't really think there's value in testing `uninterrupted` since it would be identical to the mutex test
- During the sync with Pato, we agreed that testing the default implementations would not really be worth the effort and that testing the dynamic handlers would be enough

**Pull request merged**